code/iaas/logic/src/main/java/io/cattle/platform/process/instance/InstancePreCreate.java

```diff
@@ -154,6 +154,9 @@ protected void setDns(Instance instance, Map<String, Object> labels, Map<Object,
                 data.put(DockerInstanceConstants.FIELD_DNS, dnsList);
                 data.put(InstanceConstants.FIELD_DNS_INTERNAL, Joiner.on(",").join(dnsList));
             }
+            if (labels.containsKey(SystemLabels.LABEL_RANCHER_CONTAINER_DNS_PRIORITY) && labels.get(SystemLabels.LABEL_RANCHER_CONTAINER_DNS_PRIORITY).toString().trim().equals("None")) {
```

Comment:
@kinarashah 
1) we should add handling for the case when label value is null
2) do comparison in case insensitive manner